typescript/src/ethereum.ts

```diff
@@ -132,6 +163,47 @@ export class Bridge
     super(config, BridgeDeployment)
   }
 
+  // eslint-disable-next-line valid-jsdoc
+  /**
+   * @see {ChainBridge#getDepositRevealedEvents}
+   */
+  async getDepositRevealedEvents(
+    fromBlock?: number,
+    toBlock?: number,
+    ...filterArgs: Array<any>
+  ): Promise<DepositRevealedEvent[]> {
+    const events: EthersEvent[] = await this.getEvents(
+      "DepositRevealed",
+      fromBlock,
+      toBlock,
+      ...filterArgs
+    )
+
+    return events.map<DepositRevealedEvent>((event) => {
+      return {
+        blockNumber: BigNumber.from(event.blockNumber).toNumber(),
+        blockHash: event.blockHash,
+        transactionHash: event.transactionHash,
+        fundingTxHash: TransactionHash.from(
+          event.args!.fundingTxHash
+        ).reverse(),
+        fundingOutputIndex: BigNumber.from(
+          event.args!.fundingOutputIndex
+        ).toNumber(),
+        depositor: new Address(event.args!.depositor),
+        amount: BigNumber.from(event.args!.amount),
+        blindingFactor: event.args!.blindingFactor,
+        walletPublicKeyHash: event.args!.walletPubKeyHash,
+        refundPublicKeyHash: event.args!.refundPubKeyHash,
+        refundLocktime: event.args!.refundLocktime,
+        vault:
+          event.args!.vault === constants.AddressZero
+            ? undefined
+            : new Address(event.args!.vault),
```

Comment:
Just to make sure: Are we sure that all values that come from the event conform to the specification of the `Deposit` fields as is? For example, are the byte values unprefixed?

Reply:
Yep, we need a uniform way to represent those values. We started with unprefixed strings as this was the best way to deal with such values in the code related to Bitcoin (deposit script etc.). Now, we have multiple parts that need different formats so we should definitely improve that. Can we leave a TODO near the `Deposit` type to track this?

Reply:
Ah ok, you already did it in https://github.com/keep-network/tbtc-v2/commit/13508d21680cc22c62bacefe5c6fc05069816aef. Good!